src/rgw/rgw_sal.h

```diff
@@ -673,6 +652,27 @@ class Bucket {
     // XXXX hack
     virtual void set_owner(rgw::sal::User* _owner) = 0;
 
+    /// Input parameters for create().
+    struct CreateParams {
+      std::string zonegroup_id;
+      rgw_placement_rule placement_rule;
+      // zone placement is optional on buckets created for another zonegroup
+      const RGWZonePlacementInfo* zone_placement;
+      RGWAccessControlPolicy policy;
+      Attrs attrs;
+      bool obj_lock_enabled = false;
+      std::string marker;
+      std::string bucket_id;
+      std::optional<std::string> swift_ver_location;
+      std::optional<RGWQuotaInfo> quota;
+      std::optional<ceph::real_time> creation_time;
+    };
+
+    /// Create this bucket in the backing store.
+    virtual int create(const DoutPrefixProvider* dpp,
```

Comment:
Minor nit: Bucket now has create() and remove_bucket().  I think, for consistency, this should be create() and remove() or create_bucket() and remove_bucket().

Reply:
any preference? if not, i'll go with create() and remove()